include/taco/index_notation/provenance_graph.h

```diff
@@ -111,6 +111,43 @@ struct SplitRelNode : public IndexVarRelNode {
 
 bool operator==(const SplitRelNode&, const SplitRelNode&);
 
+// DivideRelNode takes a parentVar's iteration space and divides it into divFactor
+// equal pieces. outerVar iterates over the number of pieces, and innerVar iterates
+// over each piece.
+struct DivideRelNode : public IndexVarRelNode {
+  DivideRelNode(IndexVar parentVar, IndexVar outerVar, IndexVar innerVar, size_t divFactor);
+
+  const IndexVar& getParentVar() const;
+  const IndexVar& getOuterVar() const;
+  const IndexVar& getInnerVar() const;
+  const size_t& getDivFactor() const;
+
+  void print(std::ostream& stream) const;
+  bool equals(const DivideRelNode& rel) const;
+
+  std::vector<IndexVar> getParents() const; // parentVar
+  std::vector<IndexVar> getChildren() const; // outerVar, innerVar
+  std::vector<IndexVar> getIrregulars() const; // innerVar
+
+  // computeRelativeBound performs similar logic to SplitRelNode::computeRelativeBound.
+  std::vector<ir::Expr> computeRelativeBound(std::set<IndexVar> definedVars, std::map<IndexVar, std::vector<ir::Expr>> computedBounds, std::map<IndexVar, ir::Expr> variableExprs, Iterators iterators, ProvenanceGraph provGraph) const;
```

Comment:
This line looks a bit long. (There are a few other pretty long lines in this PR that should be broken up.)

Reply:
Done